src/search.c

```diff
@@ -850,12 +855,16 @@ enum swupd_code search_main(int argc, char **argv)
 		goto clean_exit;
 	}
 
-	do_search(MoM, search_type, search_string);
+	search = do_search(MoM, search_type, search_string);
 
 clean_exit:
 	if (MoM) {
 		free_manifest(MoM);
 	}
 	swupd_deinit();
-	return ret;
+	/* if we had a failure previous to the search, return that */
+	if (ret) {
```

Comment:
when --init is used and the function is successful, ret is set to SWUPD_OK that is 0. so search is returned, and search value is SWUPD_NO.

Reply:
so do you think when using --init we should also return SWUPN_NO (1) then?

Reply:
fixed